plenum/client/wallet.py

```diff
@@ -216,6 +218,23 @@ def signOp(self,
         request = Request(operation=op, protocolVersion=CURRENT_PROTOCOL_VERSION)
         return self.signRequest(request, identifier)
 
+    def do_multi_sig_on_req(self, request: Request, identifier: str):
+        idr = self.requiredIdr(idr=identifier)
+        signature = self.signMsg(msg=request.signingState(identifier),
+                                 identifier=idr)
+        request.add_signature(idr, signature)
+
+    def sign_using_multi_sig(self, op: Dict=None, request: Request=None,
```

Comment:
I dont see the problem you are indicating, is it the extra call to `xor`, we use this pattern on lot of places in the code.

Reply:
Well, in general this is not a good pattern when API has a lot of optional parameters and encapsulates different behaviours. 
I don't have a strong opinion about this particular class since it's a client code which will be deprecated soon.